crates/client/src/storage.rs

```diff
@@ -108,6 +133,20 @@ pub trait ContentStorage: Send + Sync {
     ) -> Result<AnyHash>;
 }
 
+/// Trait for namespace map storage implementations.
+///
+/// Namespace Map storage data must be synchronized if shared between
+/// multiple threads an
+#[async_trait]
+pub trait NamespaceMapStorage: Send + Sync {
+    /// Loads namespace map
+    async fn load_namespace_map(&self) -> Result<Option<HashMap<String, String>>>;
+    /// Reset namespace mappings
+    async fn reset_namespaces(&self) -> Result<()>;
+    /// Store namespace mapping
+    async fn store_namespace(&self, namespace: String, registry_domain: String) -> Result<()>;
```

Comment:
The fact that `registry_domain` is a `String` here rather than a `HeaderValue` is another strong hint that it should have its own type.